src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private int height;
+    private int weight;
+    private double diagonal;
+
+    public Rectangle(Color color, int weight, int height) {
+        super(color);
+        this.weight = weight;
+        this.height = height;
+        calculate();
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: rectangle, area: " + getArea() + " sq. units, diagonal: "
+                + diagonal + ", height: " + height
+                + ", weight: " + weight + ", color: " + getColor());
+    }
+
+    @Override
+    public void calculate() {
```

Comment:
method comletes 2 tasks at the same time, let's splite the login into to separate methods